cli/src/lib.rs

```diff
@@ -132,6 +137,10 @@ impl Handler<iroha_core::NetworkMessage> for FromNetworkBaseRelay {
     async fn handle(&mut self, msg: iroha_core::NetworkMessage) -> Self::Result {
         use iroha_core::NetworkMessage::*;
 
+        if self.freeze_status.load(Ordering::Relaxed) > 0 {
```

Comment:
I would just use `SC`. If you think this is on a hot path then I will give it a deep thought. Ordering is hard to get right

Reply:
I agree that relaxed ordering could be a bit too much, but this is a classic use case for `Acquire`/`Release` IMO.

Reply:
I wouldn't be so reluctant to approve even `Acq/Rel`. `Relaxed` is just dangerous if there is no strong reason to use it

Reply:
Relaxed is fine in this case since the atomic is not used to guard any seperate non-atomic data. (https://www.youtube.com/watch?v=ZQFzMfHIxng&t=3824s) Since we only have one reading thread and a seperate writing thread in C you wouldn't need atomics for this. I'm simply poking a value and want the other thread to eventually see the new value. Rust requires atomics for this because of it's safety rules.

Since this isn't performance critical in any way I will change it to SeqCst in order to ease the merging of the code.

Reply:
My two cents: 
- @mversic and @ilchu , you're right, but a tad overzealous. IMO this can work with `Relaxed` just as well. We only need interior mutability. In C this would be a volatile variable. 
- @SamHSmith You're right, but a tad careless. If the ordering is not `SeqCst` and the reason isn't obvious from the same file, your justification should have been a comment in the code. 
- Not doing `SeqCst` is IMO a premature optimisation. Lock-free programming is a place where you *should* be careful.